rfcs/text/0015_nodejs_clustering.md

```diff
@@ -0,0 +1,731 @@
+We do store a state in the SavedObjects store of the last time the usage was sent to prevent sending multiple times 
+(although race conditions might occur).
+
+- Tasks storing telemetry data
+  
+We have tasks across several plugins storing data in savedobjects specifically for telemetry. Under clustering these 
+tasks will be registered multiple times.
+
+Note that sending the data multiple times doesn’t have any real consequences, apart from the additional number of ES requests, 
+so this should be considered non-blocking and only an improvement.
+
+- Event-based telemetry
+
+Event-based telemetry may be affected as well. Both the existing one in the Security Solutions team and the general 
+one that is in the works. More specifically, the size of the queues will be multiplied per worker, also growing in the
+amount of network bandwidth used, and potentially affecting our customers.
+
+We could address that by making sure that the queues are held only in the main worker.
+
+#### Task Manager
+
+Currently, task manager does "claims" for jobs to run based on the server uuid. We think this could still work with 
+a multi-process setup - each task manager in the worker would be doing "claims" for the same server uuid, which
+seems functionally the same as setting max_workers to `current max_workers * number of workers`.
+Another alternative would be to compose something like `${server.uuid}-${worker.Id}`, as TM only
+really needs a unique identifier.
+
+However, as a first step we can simply run Task Manager on the main worker. This doesn't completely solve potential
+noisy neighbor problems as the main worker will still be receiving & serving http requests, however it will at least
+ensure that other worker processes are free to serve http requests without risk of TM interference. Long term, we
+could explore manually spawning a dedicated child process for background tasks that can be called from workers, and
+thinking of a way for plugins to tell Core when they need to run things in the background.
+
+It would be ideal if we could eventually solve this with our multi-process setup, however this needs more design work
+and could necessitate an RFC in its own right. The key thing to take away here is that the work we are doing in this
+RFC would not prevent us from exploring this path further in a subsequent phase. In fact, it could prove to be a
+helpful first step in that direction.
+
+#### Alerting
+
+Currently haven't identified any Alerting-specific requirements that aren't already covered by the
+Task Manager requirements.
+
+## 6.3 Summary of breaking changes
+
+### 6.3.1 instanceUUID
+
+Depending on whether we are able to identify any cases where a shared uuid would be problematic, we may have to change
+the server.uuid configuration property when clustering mode is enabled, which would be a breaking change.
+
+At this point, we have not identified any places where this is strictly necessary.
```

Comment:
To make sure I understand it correctly: we don't need to introduce a breaking change. Do we?

Reply:
> we don't need to introduce a breaking change

Correct. Alerting uses `server.uuid` currently, but we agreed we could introduce a new ID that combines `server.uuid` and `worker.id` for their use case.

Reply:
I went ahead and removed this from the "breaking changes" section to prevent confusion.